src/python/pants/core/goals/test.py

```diff
@@ -62,56 +69,61 @@ class TestResult(EngineAwareReturnType):
     stdout_digest: FileDigest
     stderr: str
     stderr_digest: FileDigest
-    address: Address
     output_setting: ShowOutput
     result_metadata: ProcessResultMetadata | None
+    description: str
 
     coverage_data: CoverageData | None = None
     # TODO: Rename this to `reports`. There is no guarantee that every language will produce
     #  XML reports, or only XML reports.
     xml_results: Snapshot | None = None
     # Any extra output (such as from plugins) that the test runner was configured to output.
     extra_output: Snapshot | None = None
+    # Path prefix to use for the extra output above when materializing it into the final dist dir.
+    extra_output_prefix: str | None = None
```

Comment:
Instead of adding this field I thought we might want to say that plugins are responsible for adding a unique prefix to any `extra_output` they return - right now only `pytest` seems to use the extra-output feature so it's hard to gauge the trade-offs.

Reply:
cc @benjyw - fyi this is necessary because a `TestResult` might not correspond to a single address anymore, so we cannot autocompute the value like we did before.